test/Elastic.Apm.Tests/MetricsTests.cs

```diff
@@ -311,13 +311,15 @@ public void CollectGcMetrics()
 				var hasGenSize = false;
 				var hasGcTime = false;
 
+				var arrayLength = 0;
+
 				for (var j = 0; j < 10; j++)
 				{
 					for (var i = 0; i < 500; i++)
 					{
 						var array = new int[10000000];
 						// In order to make sure the line above is not optimized away, let's use the array:
-						_output.WriteLine($"GC test, int[] allocated with length: {array.Length}");
+						arrayLength += array.Length;
```

Comment:
Oh, you are right, this overflows. Honestly I did not think about that.

> But I assume the purpose here is to prevent optimization and we are not interested in the actual value.

Exactly.

> We could probably change this to numberOfAllocations and simply increment it?

I specifically wanted to use the `array` variable, because that's the one which causes the allocation, therefore I did not want it to be unused. Just incrementing a counter would still make `array` unused.

I ended up with changing it to `long` - that does not overflow.

Reply:
Thanks for the clarification. Using `long` was my initial thought as well.